CMake fails to build RFC module

The recent updates to the RFC Forecast module break CMake parallel builds

## Expected Behavior
CMake-based builds should build all model code with no errors, with any specified level of parallelism.

## Current Behavior
After configuring using CMake, then building with `make -j <n>`, the build will fail with the following error:

```
Scanning dependencies of target hydro_routing_reservoirs_rfc
[  5%] Building Fortran object Routing/Reservoirs/RFC_Forecasts/CMakeFiles/hydro_routing_reservoirs_rfc.dir/module_rfc_forecasts_properties.F.o
/Users/rcabell/work/Issues/update_cmake/trunk/NDHMS/Routing/Reservoirs/RFC_Forecasts/module_rfc_forecasts_properties.F:9:8:

    9 |     use module_reservoir, only: reservoir_properties
      |        1
Fatal Error: Cannot open module file 'module_reservoir.mod' for reading at (1): No such file or directory
compilation terminated.
make[2]: *** [Routing/Reservoirs/RFC_Forecasts/CMakeFiles/hydro_routing_reservoirs_rfc.dir/module_rfc_forecasts_properties.F.o] Error 1
make[1]: *** [Routing/Reservoirs/RFC_Forecasts/CMakeFiles/hydro_routing_reservoirs_rfc.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
```

## Your Environment
* Version of the code used: `master` branch
* Compiler and version: gfortran 8.3.0 and cmake 3.15.4

--- a/trunk/NDHMS/OrchestratorLayer/CMakeLists.txt
+++ b/trunk/NDHMS/OrchestratorLayer/CMakeLists.txt
@@ -7,4 +7,6 @@ add_library(hydro_orchestrator STATIC
         orchestrator.f90
         )
 
+add_dependencies(hydro_orchestrator hydro_utils)
+
 target_link_libraries(hydro_orchestrator PUBLIC hydro_netcdf_layer)

--- a/trunk/NDHMS/OrchestratorLayer/CMakeLists.txt
+++ b/trunk/NDHMS/OrchestratorLayer/CMakeLists.txt
@@ -7,6 +7,8 @@ add_library(hydro_orchestrator STATIC
         orchestrator.f90
         )
 
+add_dependencies(hydro_orchestrator hydro_netcdf_layer)
 add_dependencies(hydro_orchestrator hydro_utils)
 
 target_link_libraries(hydro_orchestrator PUBLIC hydro_netcdf_layer)
+target_link_libraries(hydro_orchestrator PUBLIC hydro_utils)

--- a/trunk/NDHMS/Routing/Reservoirs/RFC_Forecasts/CMakeLists.txt
+++ b/trunk/NDHMS/Routing/Reservoirs/RFC_Forecasts/CMakeLists.txt
@@ -6,5 +6,5 @@ add_library(hydro_routing_reservoirs_rfc STATIC
 	module_rfc_forecasts_properties.F
 )
 
-add_dependencies(hydro_routing_reservoirs_levelpool hydro_routing_reservoirs)
+add_dependencies(hydro_routing_reservoirs_rfc hydro_routing_reservoirs)
 

